### PR TITLE
Update: Related line chart utilities

### PR DESCRIPTION
**Pull request by kevinhinterlong**
## Description
Updates a lot of line chart related utilities to RequestV2 and typescript. Removed some references to dimension `|${field}` code that we will no longer be using. This is part 1 of at least 2 PRs for line chart

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/core/addon/chart-builders/metric.ts [hunk 1]
```diff
@@ -44,57 +47,54 @@ export default EmberObject.extend({
...
     // Support different `dateTime` formats by mapping them to a standard
-    const buildDateKey = dateTime => moment(dateTime).format(DateUtils.API_DATE_FORMAT_STRING);
+    const buildDateKey = (dateTime: MomentInput) => moment(dateTime).format(API_DATE_FORMAT_STRING);
 
-    const metrics = config.metrics;
-    const isDateTime = f => f.type === 'timeDimension' && f.field === `${request.table}.dateTime`;
-    const grain = request.columns.find(isDateTime).parameters.grain;
-    const interval = request.filters.find(isDateTime);
-    const requestInterval = Interval.parseFromStrings(interval.values[0], interval.values[1]);
+    // const metrics = config.metrics;
```
Comment (jkusa) [inline packages/core/addon/chart-builders/metric.ts:60]:
```suggestion
```
??

File: packages/core/addon/models/bard-request-v2/request.ts [hunk 2]
```diff
@@ -193,7 +193,7 @@ export default class RequestFragment extends Fragment.extend(Validations) implem
    * @property {string} timeGrain - The grain parameter of the column containing the dateTime timeDimension
    */
   @computed('timeGrainColumn.parameters.grain')
-  get timeGrain() {
+  get timeGrain(): string | undefined {
```
Comment (jkusa) [inline packages/core/addon/models/bard-request-v2/request.ts:196]: Can we use the grain type?

File: packages/core/addon/utils/duration-utils.ts [hunk 3]
```diff
@@ -21,42 +21,44 @@ const YEAR_MAP = {
...
-   * @method subtractDurationFromDate
-   * @param {Moment} date - Moment object representing date
-   * @param {Duration} duration - duration to subtract from given date
-   * @returns {Moment} - resultant date of the subtraction
+   *  @param date - Moment object representing date
+   * @param duration - duration to subtract from given date
+   * @returns resultant date of the subtraction
    */
-  subtractDurationFromDate: function(date, duration) {
+  subtractDurationFromDate(date: Moment, duration: Duration) {
```
Comment (daniellevinson) [inline packages/core/addon/utils/duration-utils.ts:38]:
```suggestion
  subtractDurationFromDate(date: Moment, duration: Duration): Moment {
```

File: packages/core/addon/components/navi-visualization-config/series-chart.js [hunk 4]
```diff
@@ -32,7 +31,7 @@ class NaviVisualizationConfigSeriesChartComponent extends Component {
    */
   @computed('request')
   get metrics() {
-    return getRequestMetrics(this.request);
+    return this.request.metricColumns;
```
Comment (daniellevinson) [inline packages/core/addon/components/navi-visualization-config/series-chart.js:34]: `@readOnly` ?

File: packages/core/addon/chart-builders/metric.ts [hunk 5]
```diff
@@ -44,57 +47,54 @@ export default EmberObject.extend({
    * @param {Object} request - request used to get data
    * @returns {Array} array of c3 data with x values
    */
-  buildData(data, config, request) {
+  buildData(data: ResponseV1['rows'], _config: unknown, request: RequestFragment) {
+    const timeGrainColumn = request.timeGrainColumn.canonicalName;
```
[the remark this answers is not in the document]
Comment (kevinhinterlong) [reply packages/core/addon/chart-builders/metric.ts:51]: No this is intended for reading out the date value from the response rows so we need the canonical name

File: packages/core/addon/chart-builders/metric.ts [hunk 6]
```diff
@@ -44,57 +47,52 @@ export default EmberObject.extend({
...
+    const interval = request.interval as Interval;
+    const timeGrain = request.timeGrain as Grain;
```
Comment (jkusa) [inline packages/core/addon/chart-builders/metric.ts:53]:
```suggestion
    const timeGrain = request.timeGrain;
```

**Review by mattgraham1995 — APPROVED** [review]

File: packages/core/addon/utils/classes/interval.ts [hunk 7]
```diff
@@ -4,31 +4,42 @@
...
 const NEXT = 'next';
-const MACROS = A([CURRENT, NEXT]);
+const MACROS = <const>[CURRENT, NEXT];
+
+function isMacro(property: unknown): property is typeof MACROS[number] {
+  const allMacros: string[] = [...MACROS];
+  return typeof property === 'string' && allMacros.includes(property);
+}
+
+type Serialized<T extends Moment | string> = { start: T; end: T | undefined };
```
Comment (jkusa) [inline packages/core/addon/utils/classes/interval.ts:22]: Do we use this for anything other than `Serialized<Moment>`?

**Review by daniellevinson — APPROVED** [review]

File: packages/core/addon/utils/duration-utils.ts [hunk 8]
```diff
@@ -21,42 +21,44 @@ const YEAR_MAP = {
...
 
 export default {
   /**
    * Subtracts a given duration from a given date
    *
-   * @method subtractDurationFromDate
-   * @param {Moment} date - Moment object representing date
-   * @param {Duration} duration - duration to subtract from given date
-   * @returns {Moment} - resultant date of the subtraction
+   *  @param date - Moment object representing date
```
Comment (jkusa) [inline packages/core/addon/utils/duration-utils.ts:34]:
```suggestion
   * @param date - Moment object representing date
```

**Review by jkusa — APPROVED** [review]

File: packages/core/addon/utils/classes/interval.ts (hunk 7 above)
Comment (kevinhinterlong) [reply packages/core/addon/utils/classes/interval.ts]: Nope good catch

**Review by jkusa — APPROVED** [review]

**Pull request merged**